keanu-project/src/main/java/io/improbable/keanu/vertices/intgr/nonprobabilistic/operators/multiple/IntegerConcatenationVertex.java

```diff
@@ -30,6 +34,21 @@ public IntegerConcatenationVertex(int dimension, IntegerVertex... input) {
         setParents(input);
     }
 
+    public IntegerConcatenationVertex(@LoadVertexParam(DIMENSION_NAME) int dimension,
+                                      @LoadVertexParam(INPUT_ARRAY_NAME) Vertex[] input) {
+        this(dimension, convertVertexArrayToIntegerVertex(input));
+    }
+
+    private static IntegerVertex[] convertVertexArrayToIntegerVertex(Vertex[] input) {
```

Comment:
return Arrays.stream(input).toArray(IntegerVertex[]::new);

Reply:
Done, and thanks for pointing this out.